File: Python/IntersectionOfTwoArrays.py
"""
Given two arrays, write a function to compute their intersection.
"""
from typing import List


class IntersectionOfTwoArrays:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        # Set for storing elements of first array
        set1 = set(nums1)
        # This set will store only common elements
        set2 = set(nums2)
        # Loop through the elements of the second array
        return list(set1 & set2)


if __name__ == "__main__":
    i = IntersectionOfTwoArrays()
    print(i.intersection([1, 2, 2, 1], [2, 2]))
    print(i.intersection([4, 9, 5], [9, 4, 9, 8, 4]))

File: JavaScript/IntersectionOfTwoArrays.js
/**
 * Given two arrays, write a function to compute their intersection.
 */
const intersectionOfTwoArrays = (nums1, nums2) => {
    // Set for storing elements of first array
    const set1 = new Set();
    // Add all the elements
    for (let i = 0; i < nums1.length; i++) {
        set1.add(nums1[i]);
    }
    // This set will store only common elements
    const set2 = new Set();
    // Loop through the elements of the second array
    for (let i = 0; i < nums2.length; i++) {
        if (set1.has(nums2[i])) {
            set2.add(nums2[i]);
        }
    }
    return Array.from(set2);
};

console.log(intersectionOfTwoArrays([1, 2, 2, 1], [2, 2]));
console.log(intersectionOfTwoArrays([4, 9, 5], [9, 4, 9, 8, 4]));